File: app/src/main/java/com/yalantis/beamazingtoday/sample/Goal.java
package com.yalantis.beamazingtoday.sample;

import android.graphics.Bitmap;

import com.yalantis.beamazingtoday.interfaces.BatModel;

import java.util.ArrayList;

/**
 * Created by lenovo on 22.08.16.
 */
public class Goal implements BatModel {

    private String name;

    private boolean isChecked;

    public boolean hasImg;

    public ArrayList<Bitmap> imgs;

    public Goal(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setChecked(boolean checked) {
        isChecked = checked;
    }

    public void setImgs(ArrayList<Bitmap> imgs){
        this.imgs=imgs;
        hasImg=true;
    }

    @Override
    public boolean hasImg() {
        return hasImg;
    }

    @Override
    public boolean isChecked() {
        return isChecked;
    }

    @Override
    public String getText() {
        return getName();
    }

}

File: app/src/main/java/com/yalantis/beamazingtoday/sample/Goal.kt
package com.yalantis.beamazingtoday.sample

import android.graphics.Bitmap

import com.yalantis.beamazingtoday.interfaces.BatModel

import java.util.ArrayList

/**
 * Created by lenovo on 22.08.16.
 */
class Goal(var name: String?) : BatModel {

    private var isChecked: Boolean = false

    var hasImg: Boolean = false

    var imgs = ArrayList<Bitmap>()
        set(imgs) {
            hasImg = true
            field = imgs
        }


    override fun setChecked(checked: Boolean) {
        isChecked = checked
    }

//    fun setImgs(imgs: ArrayList<Bitmap>) {
//        this.imgs = imgs
//        hasImg = true
//    }

    override fun hasImg(): Boolean {
        return hasImg
    }

    override fun isChecked(): Boolean {
        return isChecked
    }

    override fun getText(): String? {
        return name
    }

}
